Sources/Plasma/NucleusLib/pnAsyncCoreExe/Private/Nt/pnAceNt.cpp

```diff
@@ -312,28 +267,15 @@ void NtInitialize () {
         return;
     s_running = true;
 
-    // create a cleanup event
-    s_waitEvent = CreateEvent(
-        nullptr,
-        true,           // manual reset
-        false,          // initial state off
-        nullptr         // name
-    );
-    if (!s_waitEvent)
-        ErrorAssert(__LINE__, __FILE__, "CreateEvent {#x}", GetLastError());        
-
     // create IO completion port
     if (s_ioPort = CreateIoCompletionPort(INVALID_HANDLE_VALUE, nullptr, 0, 0); s_ioPort == nullptr)
         ErrorAssert(__LINE__, __FILE__, "CreateIoCompletionPort {#x}", GetLastError());
 
     // calculate number of IO worker threads to create
-    if (!s_pageSizeMask) {
-        SYSTEM_INFO si;
-        GetSystemInfo(&si);
-        s_pageSizeMask = si.dwPageSize - 1;
-
+    if (!s_ioThreadCount) {
         // Set worker thread count
-        s_ioThreadCount = si.dwNumberOfProcessors * 2;
+        s_ioThreadCount = std::thread::hardware_concurrency() * 2;
+        ASSERT(s_ioThreadCount);  // The standard indicates the above can return zero...
```

Comment:
O.o - I'd be tempted to `s_ioThreadCount = std::max(1, std::thread::hardware_concurrency() * 2);`